chainer/functions/array/stack.py

```diff
@@ -6,12 +6,61 @@ def stack(xs, axis=0):
     """Concatenate variables along a new axis.
 
     Args:
-        xs (list of chainer.Variable): Variables to be concatenated.
-        axis (int): Axis of result along which variables are stacked.
+        xs (list of :class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variables to be concatenated. The variables must have the
+            same shape.
+        axis (int): The axis along which the arrays will be stacked. Default
+            is 0.
```

Comment:
I am checking the specification of the negative axis in NumPy (i.e. NumPy guarantees the behavior or it happens to behave correctly), as pointed out in #2532. So, could you wait applying this change? Or I would appreciate it if you investigate it if you have a time.